trieste/models/model_interfaces.py

```diff
@@ -499,6 +511,32 @@ def evaluate_loss_of_model_parameters() -> tf.Tensor:
         multiple_assign(self.model, current_best_parameters)
 
 
+class NumDataPropertyMixin:
+    """Mixin class for exposing num_data as a property, stored in a tf.Variable. This is to work
+    around GPFlow storing it as a number, which prevents us from updating it without retracing.
+    The property is required due to the way num_data is used in the model elbo methods.
+
+    Note that this doesn't support a num_data value of None.
+
+    This should be removed once GPFlow is updated to support Variables as num_data.
+    """
+
+    _num_data: tf.Variable
+
+    @property
+    def num_data(self) -> TensorType:
+        return self._num_data.value()
+
+    @num_data.setter
+    def num_data(self, value: TensorType) -> None:
+        self._num_data.assign(value)
+
+
+class SVGPWrapper(SVGP, NumDataPropertyMixin):
+    """A wrapper around GPFlow's SVGP class that stores num_data in a tf.Variable and exposes
+    it as a property."""
+
+
```

Comment:
Do we really need this (and the VGP one below). It just complicated things. Why not always assume its a standard SVGP coming into the SParseVariational Class?

Reply:
Even if we only supported passing in standard SVGPs into SparseVariational, we'd still need this, as we need to turn those standard SVGPs into SVGPs that store num_data in a Variable and expose it as a property.

Reply:
I see